library/src/main/java/com/heinrichreimersoftware/materialintro/slide/SimpleSlide.java

```diff
@@ -519,10 +539,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                 descriptionView.setTextColor(textColorSecondary);
             }
 
+            if (getActivity() instanceof SimpleSlideActivity) {
+                ((SimpleSlideActivity)getActivity()).onSlideViewCreated(this, fragment, id);
+            }
+
             return fragment;
         }
 
         @Override
+        public void onDestroyView() {
+            if (getActivity() instanceof SimpleSlideActivity) {
+                int id = getArguments().getInt(ARGUMENT_ID);
+                ((SimpleSlideActivity)getActivity()).onSlideDestroyView(this, getView(), id);
+            }
+            innerLinearLayout = null;
+            titleView = null;
+            descriptionView = null;
+            imageView = null;
+            super.onDestroyView();
+        }
+
+        public LinearLayout getInnerLinearLayout() {
+            return innerLinearLayout;
```

Comment:
Not really necessary to reveal the internal layout to the public API. This can change soon as I'm adopting `ConstraintLayout` atm. Just have methods for the `TextView`'s and the `ImageView`.